Sources/SwiftParser/Expressions.swift

```diff
@@ -917,16 +919,37 @@ extension Parser {
     /// Parse close quote.
     let closeQuote = self.parseStringLiteralQuote(
       at: openDelimiter != nil ? .trailingRaw : .trailing,
-      text: text[closeStart...]
+      text: text
     ) ?? RawTokenSyntax(missing: openQuote.tokenKind, arena: arena)
-    text = text.dropFirst(closeQuote.byteLength)
-
+    if !closeQuote.isMissing {
+      text = text.dropFirst(closeQuote.tokenText.count)
+    }
     /// Parse closing raw string delimiter if exist.
-    let closeDelimiter = self.parseStringLiteralDelimiter(at: .trailing, text: text)
+    let closeDelimiter: RawTokenSyntax? = {
+      if let closeDelimiter = self.parseStringLiteralDelimiter(
+        at: .trailing,
+        text: text
+      ) {
+        return closeDelimiter
+      }
+
+      if let openDelimiter = openDelimiter {
+        return RawTokenSyntax(
+          kind: .rawStringDelimiter,
+          text: openDelimiter.tokenText,
+          leadingTriviaPieces: [],
+          trailingTriviaPieces: [],
+          presence: .missing,
+          arena: arena
+        )
+      }
+
+      return nil
+    }()
```

Comment:
I’m personally not a huge fan of declaring a closure that you immediately call. I would prefer a paradigm like

```
let closeDelimiter: RawTokenSyntax?
if let parsedDelimiter = self.parseStringLiteralDelimiter(…) {
  closeDelimiter = parsedDelimiter
} else if let openDelimiter = openDelimiter {
  closeDelimiter = RawTokenSyntax(… missing ...)
} else {
  closeDelimiter = nil
}
```

That matches the style we use in the rest of the parser.

Reply:
Sorry I didn't check style well.
I changed it not to use closure.